packages/react/src/useFindBy.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  findOneByFieldOperation,
+  GadgetNonUniqueDataError,
+  GadgetRecord,
+  get,
+  hydrateConnection,
+  LimitToKnownKeys,
+  Select,
+} from "@gadgetinc/api-client-core";
+import { useMemo } from "react";
+import { CombinedError, useQuery, UseQueryResponse } from "urql";
+import { FindOneFunction } from "./GadgetFunctions";
+import { OptionsType } from "./OptionsType";
+import { useStructuralMemo } from "./useStructuralMemo";
+
+/**
+ * React hook to fetch a Gadget record using the `findByXYZ` method of a given manager.
+ *
+ * @param finder `findByXYZ` function from a Gadget manager that will be used
+ * @param value field value of the record to fetch
+ * @param options options for selecting the fields in the result
+ */
+export const useFindBy = <
+  OptionsT extends OptionsType, // currently necessary for Options to be a narrow type (e.g., `true` instead of `boolean`)
+  F extends FindOneFunction<OptionsT, any, any, any>,
+  Options extends F["optionsType"]
+>(
+  finder: F,
+  value: string,
+  options?: LimitToKnownKeys<Options, F["optionsType"]>
+): UseQueryResponse<GadgetRecord<Select<Exclude<F["schemaType"], null | undefined>, Options["select"]>>> => {
+  const memoizedOptions = useStructuralMemo(options);
```

Comment:
Without this, and given how options are typically written inline as an argument, we often would have objects with new identities every time. This would mean the deps of the `useMemo` directly below would be different, creating a new plan. Similarly, the `plan.variables` object would likely be different, causing us to run a new `useQuery`.